src/ConnectedMode/ServerSentEvents/ServerSentEventSessionManager.cs

```diff
@@ -18,29 +18,109 @@
+using System.Threading;
+using Microsoft.VisualStudio.Threading;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
 using SonarLint.VisualStudio.Core.ServerSentEvents.Issues;
 using SonarLint.VisualStudio.Core.ServerSentEvents.TaintVulnerabilities;
-using SonarLint.VisualStudio.Core.Binding;
 using SonarQube.Client;
 
 namespace SonarLint.VisualStudio.ConnectedMode.ServerSentEvents
 {
-    internal interface IServerSentEventSessionManager
+    internal interface IServerSentEventSessionManager : IDisposable
     {
     }
 
     [Export(typeof(IServerSentEventSessionManager))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    internal class ServerSentEventSessionManager : IServerSentEventSessionManager
+    internal sealed class ServerSentEventSessionManager : IServerSentEventSessionManager
     {
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+        private readonly IIssueChangedServerEventSourcePublisher issueChangedServerEventSourcePublisher;
+        private readonly ISonarQubeService sonarQubeClient;
+        private readonly IServerSentEventPump eventPump;
+        private readonly ITaintServerEventSourcePublisher taintServerEventSourcePublisher;
+        private readonly IThreadHandling threadHandling;
+        private bool disposed;
+        private CancellationTokenSource sessionTokenSource;
+
         [ImportingConstructor]
         public ServerSentEventSessionManager(
             IActiveSolutionBoundTracker activeSolutionBoundTracker,
             ISonarQubeService sonarQubeClient,
+            IServerSentEventPump eventPump,
             IIssueChangedServerEventSourcePublisher issueChangedServerEventSourcePublisher,
-            ITaintServerEventSourcePublisher taintServerEventSourcePublisher)
+            ITaintServerEventSourcePublisher taintServerEventSourcePublisher,
+            IThreadHandling threadHandling)
+        {
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            this.sonarQubeClient = sonarQubeClient;
+            this.eventPump = eventPump;
+            this.issueChangedServerEventSourcePublisher = issueChangedServerEventSourcePublisher;
+            this.taintServerEventSourcePublisher = taintServerEventSourcePublisher;
+            this.threadHandling = threadHandling;
+            activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionChanged;
+        }
+
+        public void Dispose()
+        {
+            if (disposed)
+            {
+                return;
+            }
+
+            activeSolutionBoundTracker.SolutionBindingChanged -= OnSolutionChanged;
+            sessionTokenSource?.Cancel();
+            issueChangedServerEventSourcePublisher?.Dispose();
+            taintServerEventSourcePublisher?.Dispose();
+            disposed = true;
+        }
+
+        internal /*for testing*/ void OnSolutionChanged(object sender, ActiveSolutionBindingEventArgs activeSolutionBindingEventArgs)
```

Comment:
Any reason why this is internal, instead of raising the event in the test? Moq has a Raise event function, see NotificationServiceTests for example

Reply:
I decided it would be simpler to avoid triggering the events and just call the method manually

Reply:
I'm not a big fan of internal testing methods, it always feels hacky to me. Raising an event is cleaner IMO and can be easily done with a helper method in the test. WDYT?

Reply:
OK I'll change it